Async.hs

```diff
@@ -84,7 +84,7 @@ waitEither a b = atomically $
 -- <<waitAny
 waitAny :: [Async a] -> IO a
 waitAny asyncs =
-  atomically $ foldr orElse retry $ map waitSTM asyncs
+  atomically $ foldr (orElse . waitSTM) retry asyncs
```

Comment:
I'm not a fan of this refactoring, the original seems clearer and the intermediate list will be optimised away.